libraries/Bluefruit52Lib/src/BLEAdvertising.cpp

```diff
@@ -385,6 +385,21 @@ bool BLEAdvertising::start(uint16_t timeout)
   return true;
 }
 
+// first set new advertising data with "Bluefruit.Advertising.setData(data,len) or "Bluefruit.ScanResponse.setData(data,len)"
+bool BLEAdvertising::updateData()
+{
+  // gap_adv long-live is required by SD v6
+  static ble_gap_adv_data_t gap_adv =
+  {
+      .adv_data      = { .p_data = _data, .len = _count },
+      .scan_rsp_data = { .p_data = Bluefruit.ScanResponse.getData(), .len = Bluefruit.ScanResponse.count() }
+  };
+  VERIFY_STATUS( sd_ble_gap_adv_set_configure(&_hdl, &gap_adv, NULL), false );
+  VERIFY_STATUS( sd_ble_gap_adv_start(_hdl, CONN_CFG_PERIPHERAL), false );
```

Comment:
can you verify that we can just call the `set_configure()` while advertising, and the data will change without us to call `adv_start()`. Otherwise I think we loose the meaning of on-the-fly.

Reply:
Sorry for the late answer, I was on vacation with limited access.
I use the proposed changes for several weeks, so after starting advertising just calling sd_ble_gap_adv_set_configure(...) will update the data.

Reply:
I must correct my answer. After rechecking my code I was still using adv_start() inside updateData(). Without adv_start() it does not work.

As mentioned by zwells it works for updating the data by using:
Bluefruit.ScanResponse.clearData();
Bluefruit.ScanResponse.addManufacturerData(manufacturerData,sizeof(manufacturerData));

But the advertising type and length must not be changed, otherwise you will get an invalid EIR (as per nrfConnect). Unfortunately it does also not work using the proposed changes.
So for switching the scanRespanse data between e.g. manufactureName and manufactureData will not work and up to now I didn't find a solution.

Sorry for the confusion and regards

Ralf

Reply:
Sorry, I am late again, thanks @elral for your effort and PR. We close since the PR doesn't resolve the on-the-fly advertising. It is better to open a new PR later on